yt_dlp/extractor/youtube.py

```diff
@@ -3281,15 +3281,13 @@ def _extract_chapters_from_engagement_panel(self, data, duration):
             for contents in content_list)), [])
 
     def _extract_heatmap_from_player_overlay(self, data):
-        content_list = traverse_obj(data, (
-            'playerOverlays', 'playerOverlayRenderer', 'decoratedPlayerBarRenderer', 'decoratedPlayerBarRenderer', 'playerBar',
-            'multiMarkersPlayerBarRenderer', 'markersMap', ..., 'value', 'heatmap', 'heatmapRenderer', 'heatMarkers', {list}))
-        return next(filter(None, (
-            traverse_obj(contents, (..., 'heatMarkerRenderer', {
-                'start_time': ('timeRangeStartMillis', {functools.partial(float_or_none, scale=1000)}),
-                'end_time': {lambda x: (x['timeRangeStartMillis'] + x['markerDurationMillis']) / 1000},
-                'value': ('heatMarkerIntensityScoreNormalized', {float_or_none}),
-            })) for contents in content_list)), None)
+        return traverse_obj(data, (
+            'frameworkUpdates', 'entityBatchUpdate', 'mutations', ..., 'payload',
+            'macroMarkersListEntity', 'markersList', 'markers', ..., {
```

Comment:
Maybe we should explicitly filter for `MARKER_TYPE_HEATMAP`?
```suggestion
            'frameworkUpdates', 'entityBatchUpdate', 'mutations',
            lambda _, v: v['payload']['macroMarkersListEntity']['markersList']['markerType'] == 'MARKER_TYPE_HEATMAP',
            'payload', 'macroMarkersListEntity', 'markersList', 'markers', ..., {
```